packages/dev/core/src/FlowGraph/valueContainer.ts

```diff
@@ -1,11 +1,21 @@
-export type ValueSetter<T> = (value: T) => void;
-
+/**
+ * This interface represents a container for a value.
+ * the intention is that only the setValue function should
+ * be used to change the value.
+ * @experimental
+ */
 export interface ValueContainer<T> {
     value: T;
-    setValue: ValueSetter<T>;
+    setValue: (value: T) => void;
 }
 
+/**
+ * @experimental
+ * @param value
+ * @returns
+ */
 export function makeValueContainer<T>(value: T): ValueContainer<T> {
+    // question: Should we wrap this in a Proxy so it's not possible to change the value directly?

```

Comment:
If we make the interface's `value` readonly, that solves this I think?

Reply:
I'd never think of this, thanks :D